internal/server/boltdbstate/project_test.go

```diff
@@ -0,0 +1,241 @@
+package boltdbstate
+
+import (
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestProject(t *testing.T) {
+	t.Run("create and get and delete", func(t *testing.T) {
+		require := require.New(t)
+
+		s := TestState(t)
+		defer s.Close()
+
+		const projectName = "testproject"
+		const appName = "testapp"
+		// Create a project with one app
+		require.NoError(s.ProjectPut(&pb.Project{
+			Name: projectName,
+			Applications: []*pb.Application{
+				{
+					Project: &pb.Ref_Project{Project: projectName},
+					Name:    appName,
+				},
+			},
+		}))
```

Comment:
We could use a multi-app test too! And in a similar vein, it would be nice if build/deploy/release etc messages had more than one. Maybe 2-3 each just to fully test the full flow beyond a single message per resource?